src/main/datatypes/MQ2GroupType.cpp

```diff
@@ -455,6 +750,64 @@ bool MQ2GroupType::GetMember(MQVarPtr VarPtr, const char* Member, char* Index, M
 			}
 		}
 		return false;
+		
+	case GroupMembers::Furthest:
+		Dest.DWord = 0;
+		Dest.Type = pFloatType;
+
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member && member->pSpawn)
+			{
+				auto distance = Distance3DToSpawn(pLocalPC->pSpawn, member->pSpawn);
+				if(distance > furthest_dist)
+					furthest_dist = distance;
+			}
+		}
+		Dest.Float = furthest_dist;
+		return true;
+
+	case GroupMembers::FurthestMelee:
+		Dest.DWord = 0;
+		Dest.Type = pFloatType;
+
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member && member->pSpawn)
+			{
+				auto class_value = member->pSpawn->GetClass();
+				if(class_value <= TotalPlayerClasses && !ClassInfo[class_value].PureCaster)
+				{
+					auto distance = Distance3DToSpawn(pLocalPC->pSpawn, member->pSpawn);
+
+					if(distance > furthest_dist)
+						furthest_dist = distance;
+				}
+			}
+		}
+		Dest.Float = furthest_dist;
+		return true;
+
+	case GroupMembers::FurthestPureCaster:
+		Dest.DWord = 0;
+		Dest.Type = pFloatType;
+
+		for(auto& member : *pLocalPC->Group)
+		{
+			if(member && member->pSpawn)
+			{
+				auto class_value = member->pSpawn->GetClass();
+				if(class_value <= TotalPlayerClasses && ClassInfo[class_value].PureCaster)
+				{
+					auto distance = Distance3DToSpawn(pLocalPC->pSpawn, member->pSpawn);
+
+					if(distance > furthest_dist)
+						furthest_dist = distance;
+				}
+			}
+		}
+		Dest.Float = furthest_dist;
```

Comment:
Same issue here. Perhaps we can add parameters to FurthestMember to further limit the result set.